Core/src/org/sleuthkit/autopsy/geolocation/GeoFilterPanel.java

```diff
@@ -440,8 +518,14 @@ public void done() {
             }
 
             if (sources != null) {
-                for (Pair<String, DataSource> source : sources) {
-                    checkboxPanel.addElement(source.getKey(), source.getValue());
+                for (Pair<String, DataSource> source : sources.dataSources) {
+                    dsCheckboxPanel.addElement(source.getKey(), null, source.getValue());
+                }
+                for (Map.Entry<ARTIFACT_TYPE, Long> entry : sources.artifactTypes.entrySet()) {
+                    String dispName = entry.getKey().getDisplayName() + " (" + entry.getValue() + ")";
+                    String iconPath = File.separator + IconsUtil.getIconFilePath(entry.getKey().getTypeID());
+                    Icon icon = new ImageIcon(getClass().getResource(iconPath));
```

Comment:
Ah, I see. I'll look into this, thanks. Yeah the slash should be "/" and it should probably just happen in getIconFilePath. I excluded it from that because the original caller seemed to break when I added it there. But I'll dig into it more.

Reply:
I believe I have fixed this @kellykelly3. Let me know if you continue to see the NPE.